src/main/java/io/snyk/agent/logic/ReportingWorker.java

```diff
@@ -119,6 +121,15 @@ void sendIfNecessary(Supplier<UseCounter.Drain> drainMaker) throws IOException {
             return;
         }
 
+        send(drainMaker);
+
+        // intentionally at the end; this method might terminate by exception
+        this.lastSuccessfulBeacon = now;
+    }
+
+    synchronized void send(Supplier<UseCounter.Drain> drainMaker) throws IOException {
+        final byte[] prefix = jsonHeader().toString().getBytes(StandardCharsets.UTF_8);
```

Comment:
This `prefix` includes the `correlationId`, so by re-doing the work we'll end up with two different `correlationId`s. Do we care? We aren't correlating using the correlation id at the moment, and, even if we were, we probably don't care about the heartbeat being in the same correlation.

Reply:
I care, yeah. doesn't sound too critical at the moment, but it's work you've already done, would be a shame to introduce a regression here.
I'll try to fix that